components/style/values/specified/font.rs

```diff
@@ -610,6 +611,296 @@ impl Parse for FontSize {
+        /// Character Variant
+        const CHARACTER_VARIANT = 0x08;
+        /// Swash glyphs
+        const SWASH = 0x10;
+        /// Ornaments glyphs
+        const ORNAMENTS = 0x20;
+        /// Annotation forms
+        const ANNOTATION = 0x40;
+    }
+}
+
+#[derive(Clone, Debug, MallocSizeOf, PartialEq)]
+/// Set of variant alternates
+pub enum VariantAlternates {
+    /// Enables display of stylistic alternates
+    Stylistic(CustomIdent),
+    /// Enables display with stylistic sets
+    Styleset(Box<[CustomIdent]>),
+    /// Enables display of specific character variants
+    CharacterVariant(Box<[CustomIdent]>),
+    /// Enables display of swash glyphs
+    Swash(CustomIdent),
+    /// Enables replacement of default glyphs with ornaments
+    Ornaments(CustomIdent),
+    /// Enables display of alternate annotation forms
+    Annotation(CustomIdent),
+    /// Enables display of historical forms
+    HistoricalForms,
+}
+
+impl ToCss for VariantAlternates {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        match *self {
+            VariantAlternates::Swash(ref atom) => {
+                dest.write_str("swash")?;
+                dest.write_str("(")?;
+                atom.to_css(dest)?;
+                dest.write_str(")")
+            },
+            VariantAlternates::Stylistic(ref atom) => {
+                dest.write_str("stylistic")?;
+                dest.write_str("(")?;
+                atom.to_css(dest)?;
+                dest.write_str(")")
+            },
+            VariantAlternates::Ornaments(ref atom) => {
+                dest.write_str("ornaments")?;
+                dest.write_str("(")?;
+                atom.to_css(dest)?;
+                dest.write_str(")")
+            },
+            VariantAlternates::Annotation(ref atom) => {
+                dest.write_str("annotation")?;
+                dest.write_str("(")?;
+                atom.to_css(dest)?;
+                dest.write_str(")")
+            },
+            VariantAlternates::Styleset(ref vec) => {
+                dest.write_str("styleset")?;
+                dest.write_str("(")?;
+                let mut iter = vec.iter();
+                iter.next().unwrap().to_css(dest)?;
+                for c in iter {
+                    dest.write_str(", ")?;
+                    c.to_css(dest)?;
+                }
+                dest.write_str(")")
+            },
+            VariantAlternates::CharacterVariant(ref vec) => {
+                dest.write_str("character-variant")?;
+                dest.write_str("(")?;
+                let mut iter = vec.iter();
+                iter.next().unwrap().to_css(dest)?;
+                for c in iter {
+                    dest.write_str(", ")?;
+                    c.to_css(dest)?;
+                }
+                dest.write_str(")")
+            },
+            VariantAlternates::HistoricalForms => {
+                dest.write_str("historical-forms")
+            },
```

Comment:
Every `dest.write_str` write with different string for each variant so I just type them by hand.
Not sure if there's any better and graceful way to write this pattern matching.
If yes, I'm willing to update it!

cc @jdm @emilio

Reply:
I think you could `derive(ToCss)` here using `#[css(function)]` annotations.

Reply:
I'll try to implement `ToCss` for `Box<[CustomIdent]>` in another PR.
Then, I'll update this.